enhancements/agent-installer/agent-based-installer.md

```diff
@@ -0,0 +1,279 @@
+and overlaps some other installation methods in applicability. Thus, in some
+cases it will contribute to even more decisions the user has to make before
+getting started.
+
+## Design Details
+
+### Open Questions [optional]
+
+- Should the command-line tool be a subcommand of openshift-installer, or a
+  standalone binary?
+- Will the containers for the assisted components be retrieved from the same
+  release payload, or will we require the user to do additional mirroring (e.g.
+  of MCE) as well?
+
+### Test Plan
+
+The agent-based installer will be covered by end-to-end testing using virtual
+machines (in a baremetal-like configuration), automated by some variation on
+the metal platform
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts/#readme). This is
+similar to the testing employed on both the baremetal IPI and assisted
+installation flows.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+N/A
+
+#### Support Procedures
+
+N/A
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in
+`Implementation History`.
+
+## Drawbacks
+
+In disconnected environments, users will still need to mirror the release
+payload to an accessible local registry, which must be long-running (i.e. it
+continues to be available once the cluster is up).
+
+Since the installation tools cannot continue running to the completion of
+building the cluster (Node 0 must eventually be rebooted to join the cluster),
+it will necessarily not be possible to use a single API to track the entire
+process of installation.
+
+Despite the fact that RHCOS is an operating system designed exclusive to run
+OpenShift, users will still need to wait for OpenShift to be *installed* into
+it after boot. A total of three boots (which are very slow on baremetal due to
+RAM testing at startup) will still be required for each host to come up as an
+OpenShift node.
```

Comment:
What is the third boot, after booting the live ISO with the agent and then rebooting to run as a node?

Reply:
1. Boot with live ISO, install to disk
2. MCO updates ostree on disk to match machine OS content shipped in cluster
3. Boot as Node

Reply:
hmm are u sure about the 3rd one?

Reply:
@cgwalters can you confirm/deny?

Reply:
That's correct, 3 boots today.  If condensing boots 2 and 3 is important, it would be quite viable to do an "offline" update of the metal image before booting it.  There's some risks to this (e.g. suddenly we'd actually be using a potentially new Ignition version) but I think we can generally cover it in CI.